code/game/mecha/mecha.dm

```diff
@@ -836,7 +837,11 @@
 	if((obj_integrity >= max_integrity) && !internal_damage)
 		to_chat(user, "<span class='notice'>[src] is at full integrity!</span>")
 		return
+	if(repairing)
+		to_chat(user, "<span class='notice'>[src] is currently being repaired!</span>")
+		return
```

Comment:
This is not the proper way to do this. If someone walks mid repair this will likely prevent the mech from being repaired ever.

Reply:
Should be fine. If they do that then line 854 will still be hit. `I.use_tool` uses a `do_after` internally